msg/estimator_status.msg

```diff
@@ -106,7 +106,11 @@ uint16 solution_status_flags # Bitmask indicating which filter kinematic state o
 
 float32 time_slip # cumulative amount of time in seconds that the EKF inertial calculation has slipped relative to system time
 
-bool pre_flt_fail # true when estimator has failed pre-flight checks and the vehicle should not be flown regardless of flight mode
+uint8 pre_flt_fail_flags # Bitmask indicating which innovation preflight check failed
```

Comment:
How about separate booleans for these status flags? Yes it's slightly more space, but overall it's still minimal and it makes it much easier to casually review and decode these fields. The log fields become self descriptive and we don't end up with the bit definitions duplicated across multiple locations and projects. It also requires a few more instructions to encode or decode a bit rather.

For example control_mode_flags, filter_fault_flags, etc are duplicated in estimator_status.msg, ecl/EKF, flight review, the ecl analysis script, and who knows how many other tools. I was thinking about adding a new estimator_status_flags with all of these broken out as booleans and only publish on change.

This is a personal pet peeve that comes up whenever I try to debug something in the estimator beyond what we already see in flight review or the analysis script.

Reply:
+1

Reply:
I totally agree, I'll do that.